### ISSUE TITLE
Latest Lazysizes core & plugins - Sep 14, 2022

### ISSUE THREAD

**Issue by saas786** [nosilver4u/ewww-image-optimizer#284]
Lazysizes was last update almost two years ago: https://github.com/nosilver4u/ewww-image-optimizer/commit/7cf46a40267ac3f5eee7abd3853912ebb4f5e3e7

Current version is 5.3.0 and latest available version is 5.3.1, so could you please bump it?

---

unveilhooks plugin

I see you are using modified version of it, so not a quick replacement with upstream original plugin is possible.

So can you please add `addStyleScript`
Ref: https://github.com/aFarkas/lazysizes/blob/5.3.1/plugins/unveilhooks/ls.unveilhooks.js#L160

**Comment by nosilver4u** [comment]
Hi, thanks for the request. I thought I had already looked at 5.3.1, but I checked again just to be sure. There are no code changes in 5.3.1, the only modifications were to documentation/code comments which don't affect the final shipped version.

The unveilhooks change is a different story, and addStyleScript() was removed as I wasn't planning to implement that at the time--I'm still not, but maybe it wouldn't hurt to restore some of the default code for devs that want to play with it. I'll look at it further when I have time, but feel free to submit a pull request if you're so inclined :)

**Comment by nosilver4u** [comment]
Yeah, that's no problem on the testing, I have plenty of test cases on my dev site :)

**Comment by saas786** [comment]
I have also brought `data-poster` & `data-autoplay` upstream logic.

If that's okay with you, we can merge it too, can bring it close to upstream.

After reviewing your `ls.unveilhooks.js`, I am thinking I can improve it a bit more.
By keep it upstream and if possible create addon for it, to cover your custom use cases.
Only if you are interested and up for such improvement?

**Comment by nosilver4u** [comment]
That's actually not a bad idea. Then devs can take advantage of all the use cases the upstream code includes. Go ahead and give it a shot!

**Comment by nosilver4u** [comment]
While bgset is similar, it's not quite what we're doing. What we're doing is somewhat similar to the RiaS plugin, but modified to account for differences in scale/crop behavior along with some tweaks for page builders and such.
Thanks so much for looking into this, and contributing those changes!
